`first` with `ignore_nulls` in `over` incorrect data in streaming

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

from polars.testing import assert_frame_equal

df = pl.DataFrame(
    {
        "a": [1, 1, 1, 1, 2, 2, 2, 2],
        "b": pl.Series([1, 2, 3, None, None, 4, 5, 6], dtype=pl.Int32),
    }
)

result = (
    df.lazy()
    .select(pl.col("b").first(ignore_nulls=True).over("a"))
    .collect(engine="streaming")
)
expected = pl.DataFrame({"b": pl.Series([1, 1, 1, 1, 4, 4, 4, 4], dtype=pl.Int32)})
assert_frame_equal(result, expected)

```


### Log output

```shell
AssertionError: DataFrames are different (value mismatch for column "b")
[left]: shape: (8,)
Series: 'b' [i32]
[
        1
        1
        1
        1
        2
        2
        2
        2
]
[right]: shape: (8,)
Series: 'b' [i32]
[
        1
        1
        1
        1
        4
        4
        4
        4
]
```

### Issue description

`first` with `ignore_nulls` in `over` incorrect data in streaming engine.

It works as expected with `POLARS_MAX_THREADS=1`.

Original PR introducing the feature: https://github.com/pola-rs/polars/pull/25105

fyi @mcrumiller 

### Expected behavior

Result as expected.

### Installed versions

On main (e5541eb4a0).

--- a/crates/polars-core/src/frame/group_by/aggregations/dispatch.rs
+++ b/crates/polars-core/src/frame/group_by/aggregations/dispatch.rs
@@ -135,7 +135,7 @@ impl Series {
                             // All values are null, we have no first non-null.
                             None
                         } else {
-                            Some(leading_zeros)
+                            Some(first + leading_zeros)
                         }
                     })
                     .collect_ca(PlSmallStr::EMPTY)
@@ -432,7 +432,7 @@ impl Series {
                             // All values are null, we have no last non-null.
                             None
                         } else {
-                            Some(len - trailing_zeros - 1)
+                            Some(first + len - trailing_zeros - 1)
                         }
                     })
                     .collect_ca(PlSmallStr::EMPTY)

--- a/py-polars/tests/unit/operations/test_group_by.py
+++ b/py-polars/tests/unit/operations/test_group_by.py
@@ -2600,3 +2600,54 @@ def test_sorted_group_by_slice() -> None:
     assert_frame_equal(lf.slice(50, 1).collect(), expected.slice(50, 1))
     assert_frame_equal(lf.slice(20, 30).collect(), expected.slice(20, 30))
     assert_frame_equal(lf.slice(20, 30).collect(), expected.slice(20, 30))
+
+
+def test_agg_first_last_non_null_25405() -> None:
+    lf = pl.LazyFrame(
+        {
+            "a": [1, 1, 1, 1, 2, 2, 2, 2, 2],
+            "b": pl.Series([1, 2, 3, None, None, 4, 5, 6, None]),
+        }
+    )
+
+    # first
+    result = lf.group_by("a", maintain_order=True).agg(
+        pl.col("b").first(ignore_nulls=True)
+    )
+    expected = pl.DataFrame(
+        {
+            "a": [1, 2],
+            "b": [1, 4],
+        }
+    )
+    assert_frame_equal(result.collect(), expected)
+
+    result = lf.with_columns(pl.col("b").first(ignore_nulls=True).over("a"))
+    expected = pl.DataFrame(
+        {
+            "a": [1, 1, 1, 1, 2, 2, 2, 2, 2],
+            "b": [1, 1, 1, 1, 4, 4, 4, 4, 4],
+        }
+    )
+    assert_frame_equal(result.collect(), expected)
+
+    # last
+    result = lf.group_by("a", maintain_order=True).agg(
+        pl.col("b").last(ignore_nulls=True)
+    )
+    expected = pl.DataFrame(
+        {
+            "a": [1, 2],
+            "b": [3, 6],
+        }
+    )
+    assert_frame_equal(result.collect(), expected)
+
+    result = lf.with_columns(pl.col("b").last(ignore_nulls=True).over("a"))
+    expected = pl.DataFrame(
+        {
+            "a": [1, 1, 1, 1, 2, 2, 2, 2, 2],
+            "b": [3, 3, 3, 3, 6, 6, 6, 6, 6],
+        }
+    )
+    assert_frame_equal(result.collect(), expected)

